src/aggregate/aggregate_plan.c

```diff
@@ -261,94 +261,108 @@ void AGPLN_Dump(const AGGPlan *pln) {
   }
 }
 
-array_t AGPLN_Serialize(const AGGPlan *pln) {
-#define APPEND_STR(s)             \
-  {                               \
-    char *s__ = strdup(s);        \
-    arr = array_append(arr, s__); \
+typedef char **myArgArray_t;
+
+static void append_string(myArgArray_t *arr, const char *src) {
+  char *s = strdup(s);
+  *arr = array_append(*arr, s);
+}
+static void append_uint(myArgArray_t *arr, unsigned long long ll) {
+  char s[64] = {0};
+  sprintf(s, "%llu", ll);
+  append_string(arr, s);
+}
```

Comment:
@mnunberg you forgot to specify inline

Reply:
Inline isn't necessary here. The biggest use of inline is inside a header file, so the compiler doesn't throw an "unused definition" warning. gcc, clang, etc. will optimize functions which are small enough as inline or not regardless of whether the `inline` keyword is used. Likewise, the `inline` keyword will be ignored if the function cannot be inlined for whatever reason.

Reply:
Still I think its a good practice to put the inline, you never know which compiler is been used and which version ..